src/main/java/org/intellij/sdk/codesync/Utils.java

```diff
@@ -45,6 +45,16 @@ public static Boolean shouldSkipEvent(String repoPath) {
         }
     }
 
+    public static boolean isIndividualFileOpen(Project project) {
+
+        VirtualFile[] virtualFiles = ProjectUtils.getAllContentRoots(project);
+        if (virtualFiles[0].getPath().equals(project.getBasePath())) {
```

Comment:
What if we have something like 
```
mydir/
    mydir (A file but having same name as parent directory)
    anotherFile.txt
```

Reply:
Will be great to add some unit tests for this function with possible edge cases

Reply:
> What if we have something like
> 
> ```
> mydir/
>     mydir (A file but having same name as parent directory)
>     anotherFile.txt
> ```

It will still work because if we are opening a single file as project on Intellij, it creating its base directory in temp directory of system. Otherwise it is same as root directory of folder.

Reply:
I understand that but the example I shared is different and seems like your logic will fail on it. In my example I have a directory which has 2 files, 1 has same name as directory, other file has different name. In this case 
```
virtualFiles[0].getPath().equals(project.getBasePath())
```
will return true, right? Since you are only comparing 1st file path with directory path. 
I think we need additional check as 
```
virtualFiles.length == 1 && virtualFiles[0].getPath().equals(project.getBasePath())
```
Thoguhts?

Reply:
No, I'm not comparing file path! The method I'm using returns content root, which is basically root directory for the folder. I will update the code and make it clear.

Reply:
I am talking about multi-modules, for that I think we need to check length of contentRoots, if there are multiple modules open, it should return false. 
Unit tests will defineltiy catch these issues before hand. Kindly add a few unit tests for different cases, thanks

Reply:
Thoughts? @saleem-latif